LIPS/lip-17.md

```diff
@@ -0,0 +1,350 @@
+---
+lip: 17
+title: MEV-Boost relays whitelist for Lido
+status: Draft
+author: George Avsetsyn, Artem Veremeenko, Eugene Mamin, Isidoros Passadis
+discussions-to: TBA
+created: 2022-08-30
+updated: 2022-09-02
+---
+
+# MEV-Boost relays whitelist for Lido
+
+>DISCLAIMER. [MEV extraction policy](https://research.lido.fi/t/discussion-draft-for-lido-on-ethereum-block-proposer-rewards-policy/2817
+) for Lido DAO is an ongoing topic and has not been finalized yet. This proposal is about additional tech spares.
+
+## Simple Summary
+
+The on-chain relays whitelist is planned to be used by Node Operators participating in the Lido protocol after the Merge to extract MEV according to the expected Lido policies.
+
+## Motivation
+
+It's proposed that Node Operators should use [`MEV-Boost`](https://github.com/flashbots/mev-boost) infrastructure developed by Flashbots to support MEV extraction through the open market mechanics as a current PBS solution that has a market fit.
+
+The proposed whitelist is intended to be a source of truth for the set of possible relays allowed to be used by Node Operators. In particular, Node Operators would use the contract to keep their software configuration up-to-date (setting the necessary relays once Lido DAO updates the set).
+
+## Mechanics
+
+The contract represents simple registry storage.
+
+Anyone can access the storage in a permissionless way through the disclosed `view` methods or by collecting the emitted events covering all storage modifications.
+
+Any modification of the set (i.e., internal storage modification) is allowed only by [the general Lido DAO governance process](https://lido.fi/governance#regular-process), which is implemented on-chain through the Aragon voting from the initial deployment stage. However, it's still possible to assign a dedicated management entity for adding/removing relay items.
+
+The proposed `MEVBoostRelayWhitelist` contract is non-upgradable, and its immutable owner is intended to be initialized with the [Lido DAO Aragon Agent](https://etherscan.io/address/0x3e40D73EB977Dc6a537aF587D48316feE66E9C8c) address on mainnet upon the deployment phase.
+
+The dedicated management `manager` entity is initialized with a zero address and can be set later by the contract's owner to be able to add/remove relays. The possible candidate to bear such responsibilities without sacrificing governance security is [Easy Track](https://easytrack.lido.fi/) (in this case  [`EVMScriptExecutor (Easy Track)`](https://docs.lido.fi/deployed-contracts#easy-track) should be set as `manager`).
+
+### Relay information structure
+
+It's proposed to use the following structure:
+
+```vyper
+struct Relay:
+    uri: String[MAX_STRING_LENGTH]
+    operator: String[MAX_STRING_LENGTH]
+    is_mandatory: bool
+    description: String[MAX_STRING_LENGTH]
+```
+
+where:
+
+- `uri` is the relay's URI to fetch the data
+- `operator` is the relay title or pseudonym
```

Comment:
I thought it is a name of the entity, running the relay, isn't it? And something like "One of the main relays of FlashBots infrastructure" goes to description.

Reply:
Agree, thx!